test/integration/targets/expect/tasks/main.yml

```
@@ -241,3 +241,106 @@
     that:
     - "non_zero_result.failed"
     - "non_zero_result.msg == 'non-zero return code'"
+
+- name: run tests in venv
+  vars:
+    venv_dir:  "{{ output_dir }}/pexpected_venv"
+    venv_python: "{{ venv_dir }}/bin/python"
+  block:
+    - name: ensure venv exists
+      pip:
+        name: coverage
+        virtualenv: "{{ venv_dir }}"
+        virtualenv_command: "{{ ansible_python_interpreter }} -m venv"
+
+    - name: trigger pexpect dependency absence
+      expect:
+        command: "echo expectThePexpected"
+        responses:
+          foo: bar
+      vars:
+        ansible_python_interpreter: "{{ venv_python }}"
+      register: pexpect_dependency_absence
+      ignore_errors: true
+
+    - name: assert pexpect dependency absence
+      assert:
+        that:
+          - "pexpect_dependency_absence.failed"
+          - "'pexpect' in pexpect_dependency_absence.msg"
```

YAML quoting is "fun" (this coming from a maintainer of PyYAML :laughing:)- most Jinja expression strings (including those that were already in the test) don't actually need outer quotes, or can be trivially rewritten not to:

```suggestion
          - pexpect_dependency_absence.failed
          - pexpect_dependency_absence.msg is contains 'pexpect'
```

LMK sometime if you want more info on when/why things *do* need to be quoted in YAML.